### PR TITLE
Chainquery integration

### PR DESCRIPTION
**Pull request by marcdeb1**
In this version, the block explorer is directly connected to Chainquery to retrieve all data.
What is changed here is mainly the SQL queries because of the different structures between Chainquery data and the block explorer database. Now the block sync process becomes useless, but the local database is still used to keep Price History.
There may still be some bugs, please share them here so that I can fix them.

Some data is inaccurate, such as the block nonces for first blocks, but it's reflecting Chainquery's data, which is now the reference.
Issues due to data inaccuracy such as #3, #45, #8, #31 are now solved.

Also, it seems that the circulating supply is closer to 200M and not 212M like shown in the previous version. This is because of a repeated address in the old calculation (https://github.com/lbryio/block-explorer/blob/3eab8389671328113badb0e315fb2bfa37fb7cea/src/Controller/MainController.php#L905) and wrong balances for addresses.

### CODE REVIEW

**Comment by tzarebczan** [comment]
The repeated address would have made the circulating supply smaller, not larger. Also, balance issues only on those addresses would have affected this. 

What's the total supply look like in comparison to `584,781,965` (current). 200M does sound closer if subtracting our current reserved balance from the 584M figure, given that we have a few million LBC on hot wallets. 

@tiger5226 do you think this is ready to go as is?

**Comment by nikooo777** [comment]
Could you please rebase onto master to get the latest changes in? I'm deploying this on a test server

**Comment by nikooo777** [comment]
you can see a running version here: https://beta.explorer.lbry.io/

**Comment by marcdeb1** [comment]
Thanks for the beta link, looks good !
@tzarebczan for the circulating supply it is close to 200M (584 - 381).
It is negative because total amount is taken from redis that might not be setup on the server. 381M is the total balance in reserved addresses.

I will check what's wrong with the blocksize chart.

**Comment by nikooo777** [comment]
Yes redis isn't configured on the server for this beta. I left the config default to use "file" rather than redis. shouldn't that work too?

**Comment by tzarebczan** [comment]
Network on mainpage shows NA, should show GH/s.

**Comment by tzarebczan** [comment]
When searching for "one" , the controlling claim (https://beta.explorer.lbry.io/claims/008401a48cecef2a13cf5376b696169bb36b18c6) doesn't show. We should show controlling first, mark as such, and then allow sorting of rest:

**Comment by marcdeb1** [comment]
- Added Controlling claim tag and sorting by Controlling claim.
- There are several data sources in the explorer : 
  - Chainquery (new)
  - API via JSON RPC (url in config/lbry.php) : used to get network hashrate and total LBC in circulation (nothing changed)
  - Redis to cache LBC price (nothing changed)
  - Local DB to store price history, only used for blocksize chart. I removed price line in the chart for the moment. To set local db, add details in config/app.php, where 'default' db is chainquery, and 'localdb' is the old one.

**Comment by nikooo777** [comment]
I'm having issues deploying this last version. is it working?

**Comment by marcdeb1** [comment]
Hey Niko, what are the issues ?

**Comment by marcdeb1** [comment]
Hey Niko, what are the issues ?

**Comment by tzarebczan** [comment]
@shyba noticed that going to https://beta.explorer.lbry.io/claims?page=6360 (last page), takes a really long time to load. Not sure why that would be - maybe pagination not being called correctly somewhere?

**Comment by marcdeb1** [comment]
Please check if there are any issues.
I cannot get Travis to work (For PHP 7.0, check is failing because some packages require 7.1, for 7.1 it's because of the database name chainquery.lbry.io that cannot be parsed).

**Comment by marcdeb1** [comment]
Awesome ! And it's really fast too

**Comment by tiger5226** [comment]
Very Fast!!! Very impressive! This is really exciting! I think this was the last block explorer item blocking this(besides CI). The rest are Chainquery items. 

For the CI failures, I think we just need to run the composer updates for the `./composer.json`. Then it should be able to find the combination of packages. It sounds like it hit a deadlock on package dependencies. @akinwale Can you take a look at the CI logs? We should make sure this is fixed before we merge.

**Comment by nikooo777** [comment]
Please let me know when this is ready to be merged/deployed so I can coordinate.

File: config/app.default.php [hunk 1]
```diff
@@ -257,7 +257,28 @@
              */
             //'init' => ['SET GLOBAL innodb_stats_on_metadata = 0'],
 
-            'url' => env('DATABASE_URL', null),
+            'url' => env('DATABASE_URL', 'chainquery.lbry.io:3600'),
```
Comment (akinwale) [inline config/app.default.php]: @marcdeb1 Please change the DSN string to `mysql://chainquery.lbry.io:3600` as this is causing the Travis build to fail. Thanks.

**Comment by akinwale** [comment]
@nikooo777 This can be merged if there are no other changes required. The Travis build isn't a blocker.
cc @tiger5226 @tzarebczan

**Comment by tiger5226** [comment]
@marcdeb1 For the slow query it is because offset is being used. see here https://www.eversql.com/faster-pagination-in-mysql-why-order-by-with-limit-and-offset-is-slow/

We could adjust it to use a seek strategy, but I dont think this is a blocker as the original takes 3 times as long.

**Comment by nikooo777** [comment]
alright I merged it, i'm deploying too. Please continue with the non blocking issues if any

**Pull request merged**